modules/snmp/validators.go

```diff
@@ -0,0 +1,144 @@
+package snmp
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+)
+
+func appendError(err error, msg string) error {
+	var e error
+	if err == nil {
+		if msg == "" {
+			return nil
+		}
+		e = errors.New(msg)
+	} else {
+		if msg == "" {
+			return err
+		}
+		e = fmt.Errorf("%w %s", err, msg)
+	}
+	return e
+}
+
+func (d Dimension) validateConfig(index_chart, index int) error {
+	var err error
+	err = nil
+	if d.Name == "" {
+		err = appendError(err, fmt.Sprintf("invalid or missing value: charts[%d].dimension[%d].name;", index_chart, index))
+	}
+	if d.OID == "" {
+		err = appendError(err, fmt.Sprintf("missing value: charts[%d].dimension[%d].oid;", index_chart, index))
+	}
+	if d.Algorithm == "" ||
+		(d.Algorithm != string(module.Incremental) &&
+			d.Algorithm != string(module.PercentOfIncremental) &&
+			d.Algorithm != string(module.PercentOfAbsolute)) {
+		err = appendError(err, fmt.Sprintf("invalid or missing value: charts[%d].dimension[%d].algorithm;", index_chart, index))
+	}
+	if d.Multiplier == 0 {
+		err = appendError(err, fmt.Sprintf("integer set to 0: charts[%d].dimension[%d].multiplier;", index_chart, index))
+	}
+	if d.Divisor == 0 {
+		err = appendError(err, fmt.Sprintf("integer set to 0: charts[%d].dimension[%d].divisor;", index_chart, index))
+	}
```

Comment:
An empty/unset value is ok here (Netdata will use an internal default), it is better to have as less mandatory parameters as possible. 0 means no divisor.

Reply:
same as multiplier comment